src/deroff.rs

```diff
@@ -538,12 +538,311 @@ impl Deroffer {
+                } else {
+                    return false;
+                }
+            } else {
+                return false;
+            }
+
+            if self.reg_table.contains_key(&reg) {
+                // This unwrap is safe because of the if
+                self.s = self.reg_table.get(&reg).unwrap().to_owned();
+                // This is unimplemented, so tests fail:
+                // self.text_arg();
+                return true;
+            } else {
+                return false;
+            }
+        } else {
+            return false;
+        }
+    }
+
+    fn digit(&self, _: usize) -> bool {
+        unimplemented!()
+    }
+
+    fn size(&mut self) -> bool {
+        /* # We require that the string starts with \s */
+        if self.digit(2) || ("-+".contains(Self::str_at(&self.s, 2)) && self.digit(3)) {
+            // self.skip_char(3);
+            self.s = self.skip_char(&self.s, Some(3)).to_owned();
+            while self.digit(0) {
+                // self.skip_char(1);
+                self.s = self.skip_char(&self.s, None).to_owned();
+            }
+            true
+        } else {
+            false
+        }
+    }
+
+    fn condputs(&self, _: &str) {
+        unimplemented!()
+    }
+
+    fn esc(&mut self) -> bool {
+        /* # We require that the string start with backslash */
+        if let Some(c) = self.s.get(1..2) {
+            match c {
+                "e" | "E" => self.condputs("\\"),
+                "t" => self.condputs("\t"),
+                "0" | "~" => self.condputs(" "),
+                "|" | "^" | "&" | ":" => (),
+                _ => self.condputs(c),
+            };
+            // self.skip_char(2);
+            self.s = self.skip_char(&self.s, Some(2)).to_owned();
+            true
+        } else {
+            false
+        }
+    }
+
+    fn word(&mut self) -> bool {
+        let mut got_something = false;
+        while let Some(m) = self.g_re_word.find(&self.s) {
+            got_something = true;
+            self.condputs(m.as_str());
+            // self.skip_char(m.end());
+            self.s = self.skip_char(&self.s, Some(m.end())).to_owned();
+
+            while self.spec() {
+                if !self.specletter {
+                    break;
+                }
+            }
+        }
+        got_something
+    }
+
+    fn text(&mut self) -> bool {
+        loop {
+            if let Some(idx) = self.s.find("\\") {
+                self.condputs(&self.s[..idx]);
+                // self.skip_char(idx);
+                if !self.esc_char_backslash() {
+                    // self.condputs(self.str_at(0));
+                    self.condputs(Self::str_at(&self.s, 0));
+                    // self.skip_char(1);
+                    self.s = self.skip_char(&self.s, None).to_owned();
+                }
+            } else {
+                self.condputs(&self.s);
+                self.s = String::new();
```

Comment:
Why not?
```rust
self.condputs(&self.s.drain(..).collect());
```

Reply:
Oh also, this
```
error[E0502]: cannot borrow `self.s` as mutable because it is also borrowed as immutable
   --> src/deroff.rs:933:32
    |
933 |                 self.condputs(&self.s.drain(..).collect::<String>());
    |                 ---- --------  ^^^^^^^^^^^^^^^^ mutable borrow occurs here
    |                 |    |
    |                 |    immutable borrow later used by call
    |                 immutable borrow occurs here
```